package.json

```diff
@@ -48,21 +48,22 @@
   "homepage": "https://github.com/jaredpalmer/react-fns#readme",
   "devDependencies": {
     "@types/jest": "20.0.6",
-    "@types/node": "^8.0.22",
-    "@types/prop-types": "15.5.1",
+    "@types/node": "^9.3.0",
+    "@types/prop-types": "15.5.2",
     "@types/qs": "^6.5.0",
-    "@types/react": "16.0.2",
-    "@types/react-dom": "15.5.1",
-    "@types/react-test-renderer": "15.5.2",
+    "@types/react": "*",
```

Comment:
The biggest headache I ran into with this PR was a TypeScript compilation error. It turns out when I upgraded to the latest version of `@types/react`, which is 16.0.35 at the time of this comment, yarn resolved two different `@types/react` packages, and then TypeScript wasn't picking up the latest type definition in its compilation process.

I believe this is due to other type definitions listing `"@types/react": "*"` as a dependency. Here's a snippet of `yarn.lock` from this PR:

```
"@types/react-dom@16.0.3":
  version "16.0.3"
  resolved "https://registry.yarnpkg.com/@types/react-dom/-/react-dom-16.0.3.tgz#8accad7eabdab4cca3e1a56f5ccb57de2da0ff64"
  dependencies:
    "@types/node" "*"
    "@types/react" "*"

"@types/react-test-renderer@16.0.0":
  version "16.0.0"
  resolved "https://registry.yarnpkg.com/@types/react-test-renderer/-/react-test-renderer-16.0.0.tgz#905a12076d7315eb4a36c4f0e5e760c7b3115420"
  dependencies:
    "@types/react" "*"
```

See https://github.com/DefinitelyTyped/DefinitelyTyped/issues/21242 for some info on the error I was running into. I don't usually like using `*` as a package version, but I'm not sure what to do here since I was unable to figure out how to resolve this. Having a `yarn.lock` file will at least resolve the version listed in the lockfile, so perhaps this isn't an issue.